contracts/Reserve/AmmReserve/AmmReserve.cpp

```diff
@@ -20,29 +20,48 @@ ACTION AmmReserve::init(name    owner,
     eosio_assert(!state_inst.exists(), "init already called");
 
     state new_state;
-    new_state.owner = owner;
+    new_state.admin = admin;
     new_state.network_contract = network_contract;
     new_state.token_symbol = token_symbol;
     new_state.token_contract = token_contract;
     new_state.eos_contract = eos_contract;
     new_state.trade_enabled = enable_trade;
-    new_state.collected_fees_in_tokens.symbol = token_symbol;
-    new_state.collected_fees_in_tokens.amount = 0;
+    new_state.collected_profit_in_tokens.symbol = token_symbol;
+    new_state.collected_profit_in_tokens.amount = 0;
     state_inst.set(new_state, _self);
 }
 
+ACTION AmmReserve::quickset(double p) {
+    auto state_inst = get_state_assert_admin();
+    params_type params_inst(_self, _self.value);
+    params new_params;
+
+    new_params.p_min = p / 2.0;
+    new_params.max_eos_cap_buy = asset(MAX_AMOUNT, EOS_SYMBOL);
+    new_params.max_eos_cap_sell = asset(MAX_AMOUNT, EOS_SYMBOL);
+    new_params.profit_percent = 0.0;
+    new_params.max_sell_rate = p * 2.0;
+    new_params.min_sell_rate = p / 2.0;
+    new_params.max_buy_rate = 1.0 / new_params.min_sell_rate;
+    new_params.min_buy_rate = 1.0 / new_params.max_sell_rate;
+
+    /* (p/p_min) = 2.0 = e^(rE) => r = ln(2)/E */
+    asset eos_balance = get_balance(_self, state_inst.get().eos_contract, EOS_SYMBOL);
+    new_params.r = 0.69314 / amount_to_damount(eos_balance.amount, EOS_PRECISION);
+
+    params_inst.set(new_params, _self);
```

Comment:
consider calling set params here? though not sure if it could work with the admin auth